warehouse/index-stats/src/main/java/datawave/mapreduce/shardStats/StatsHyperLogReducer.java

```diff
@@ -0,0 +1,117 @@
+    
+    // debug/diagnostics values
+    // total number of keys written to output table
+    private int totalKeys;
+    // interval for processing values for a single input key
+    private int valueInterval;
+    // minimum total count required to create an entry for a field name/datatype pair
+    private int minCount;
+    // produce counts only - do not write to table
+    private boolean countsOnly;
+    
+    // hyperlog properties
+    private int normalPrecision;
+    private int sparsePrecision;
+    
+    // timestamp for bulk ingest key
+    private long timestamp;
+    
+    @Override
+    public void setup(Configuration conf) throws IOException, InterruptedException {
+        super.setup(conf);
+        
+        // set log level if configured
+        String logLevel = conf.get(STATS_REDUCER_LOG_LEVEL, StatsJob.DEFAULT_LOG_LEVEL.toString());
+        Level level = Level.toLevel(logLevel, Level.INFO);
+        log.info("log level set to " + level.toString());
+        
+        this.valueInterval = conf.getInt(STATS_REDUCER_VALUE_INTERVAL, DEFAULT_VALUE_INTERVAL);
+        log.info("log value interval(" + this.valueInterval + ")");
+        
+        this.minCount = conf.getInt(STATS_MIN_COUNT, DEFAULT_MIN_COUNT);
+        log.info("minimum count(" + this.minCount + ")");
+        
+        this.countsOnly = conf.getBoolean(STATS_REDUCER_COUNTS, false);
+        log.info("counts only(" + this.countsOnly + ")");
+        
+        // hyperlog precision
+        this.normalPrecision = conf.getInt(StatsJob.HYPERLOG_NORMAL_OPTION, StatsJob.HYPERLOG_NORMAL_DEFAULT_VALUE);
+        log.info("hyperlog normal precision(" + this.normalPrecision + ')');
+        this.sparsePrecision = conf.getInt(StatsJob.HYPERLOG_SPARSE_OPTION, StatsJob.HYPERLOG_SPARSE_DEFAULT_VALUE);
+        log.info("hyperlog sparse precision(" + this.sparsePrecision + ')');
+        
+        this.timestamp = System.currentTimeMillis();
+    }
+    
+    @Override
+    public void finish(TaskInputOutputContext<?,?,BulkIngestKey,Value> context) throws IOException, InterruptedException {
+        log.info("reduce total(" + this.totalKeys + ")");
+        super.finish(context);
+    }
+    
+    @Override
+    public void doReduce(BulkIngestKey key, Iterable<Value> values, TaskInputOutputContext<?,?,BulkIngestKey,Value> context) throws IOException,
+                    InterruptedException {
+        log.info("reduce key(" + key.toString() + ")");
+        this.totalKeys++;
+        HyperLogLogPlus hllp = new HyperLogLogPlus(this.normalPrecision, this.sparsePrecision);
+        HyperLogFieldSummary stats = new HyperLogFieldSummary(hllp);
+        int valueCount = 0;
+        for (Value val : values) {
+            stats.add(val);
+            valueCount++;
+            if (0 == (valueCount % this.valueInterval) || this.countsOnly) {
+                if (this.countsOnly) {
+                    StatsHyperLogSummary addStats = new StatsHyperLogSummary(val);
```

Comment:
It looks like the intent of this block is for debugging. I'd suggest we include `&& log.isDebugEnabled()` to the if condition above. No point in generating these objects if we aren't debug logging.

Reply:
The variable countsOnly is a debug option that provides this functionality. It can be used to examine the data without creating an rfile. However, it does require the log level set to debug. The log level should be info, which is really the intent.